src/Compilers/Core/Portable/SourceGeneration/ISyntaxHelper.cs

```diff
@@ -60,5 +63,7 @@ internal abstract class AbstractSyntaxHelper : ISyntaxHelper
 
         public abstract void AddAliases(GreenNode node, ArrayBuilder<(string aliasName, string symbolName)> aliases, bool global);
         public abstract void AddAliases(CompilationOptions options, ArrayBuilder<(string aliasName, string symbolName)> aliases);
+
+        public abstract bool ContainsGlobalAliases(SyntaxNode root, CancellationToken cancellationToken);
```

Comment:
Not part of your change, but wondering why we have both interface and abstract class flavors of this? Looks like the abstract class doesn't really add functionality here?

Reply:
One developer's *awesome feature that makes it clear when a method implements an interface method* is another developer's *annoying requirement to make me restate things that can be implicit* ;)